### PR TITLE
Add logging for task stop operations

### PR DESCRIPTION
**Pull request by findingrish**

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/worker/executor/ExecutorLifecycle.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ public void start() throws InterruptedException
             }
 
             // Kind of gross, but best way to kill the JVM as far as I know
-            log.info("Triggering JVM shutdown.");
+            log.info("Task JVM is being killed by the Overlord.");
```
Comment (abhishekagarwal87) [inline indexing-service/src/main/java/org/apache/druid/indexing/worker/executor/ExecutorLifecycle.java:159]:
```suggestion
            log.info("Triggering JVM shutdown. Check overlord logs to see why the task is being shut down.");
```

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskClientAsyncImpl.java [hunk 2]
```diff
@@ -145,10 +145,16 @@ public ListenableFuture<Boolean> stopAsync(final String id, final boolean publis
...
     return makeRequest(id, new RequestBuilder(HttpMethod.POST, "/stop" + (publish ? "?publish=true" : "")))
         .onSuccess(r -> true)
-        .onHttpError(e -> Either.value(false))
-        .onNotAvailable(e -> Either.value(false))
+        .onHttpError(e -> {
+          log.warn("Task [%s] coundln't be stopped because of http request failure.", id);
+          return Either.value(false);
+        })
+        .onNotAvailable(e -> {
+          log.warn("Task [%s] coundln't be stopped because it is not available.", id);
```
[the remark this answers is not in the document]
Comment (findingrish) [reply indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskClientAsyncImpl.java:153]: this corresponds to `ServiceNotAvailableException` which is thrown when the service locator returns empty set of locations

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskClientSyncImpl.java [hunk 3]
```diff
@@ -85,7 +85,11 @@ private boolean stop(final String id, final boolean publish)
           publish ? "publish=true" : null,
           true
       );
-      return isSuccess(response);
+      boolean isSuccess = isSuccess(response);
+      if (!isSuccess) {
+        log.warn("Task [%s] coundln't be stopped because of http request failure.", id);
```
Comment (abhishekagarwal87) [inline indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskClientSyncImpl.java:90]: is it possible to get the exact http request failure?

Comment (findingrish) [reply indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskClientSyncImpl.java]: Added

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**